version: '3.8'

services:
  django-app:
    image: llodia221/gestion-doc:latest
    container_name: django_app
    ports:
      - "8000:8000"
    volumes:
      - db_data:/app/gestion_doc
    environment:
      - PYTHONUNBUFFERED=1
    restart: always
    networks:
      - gestion-doc-network
    command: >
      sh -c "cd /app/gestion_doc &&
             python manage.py migrate &&
             python manage.py runserver 0.0.0.0:8000"

  sqliteweb:
    image: tomdesinto/sqliteweb
    container_name: sqlite_explorer
    command: db.sqlite3
    volumes:
      - db_data:/db
    ports:
      - "8080:8080"
    restart: unless-stopped
    networks:
      - gestion-doc-network
    depends_on:
      - django-app

  nginx:
    image: nginx:latest
    container_name: nginx_proxy
    ports:
      - "80:80"
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf:ro
    depends_on:
      - django-app
    networks:
      - gestion-doc-network
    restart: unless-stopped

  watchtower:
    image: containrrr/watchtower
    container_name: watchtower_django
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      - WATCHTOWER_CLEANUP=true
      - WATCHTOWER_ROLLING_RESTART=true
      - WATCHTOWER_POLL_INTERVAL=300
    restart: unless-stopped
    networks:
      - gestion-doc-network

volumes:
  db_data:

networks:
  gestion-doc-network:
    driver: bridge
